ArduPlane/Plane.h

```diff
@@ -233,6 +233,9 @@ class Plane : public AP_Vehicle {
 
     // are we currently in long failsafe but have postponed it in MODE TAKEOFF until min level alt is reached
     bool long_failsafe_pending;
+    
+    //flag that we have already called autoenable fences once in MODE TAKEOFF
+    bool have_autoenabled_fences;
```

Comment:
I think this should be an attribute of fence not plane.

Reply:
The logic is just too convoluted to make that feasible